From 11f91ca11e4fd9c731ef149b41b0cb716e88f50d Mon Sep 17 00:00:00 2001
From: Eddie Huang <eddie.huang@mediatek.com>
Date: Mon, 16 Nov 2015 13:18:11 +0800
Subject: [PATCH 1/4] ARM64: dts: Mediatek: mt8173-evb: add USB power domain to
 pwrap

MT8173 E1 has one bug that if turn off USB power domain, infra
power will also be off. Pwrapper is in infra, so add USB power
domain in pwrapper

Change-Id: I3fa54425c8738c9de3349a8f47b58c39f0a0a76d
Signed-off-by: Eddie Huang <eddie.huang@mediatek.com>
---
 arch/arm64/boot/dts/mediatek/mt8173-evb.dts | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm64/boot/dts/mediatek/mt8173-evb.dts b/arch/arm64/boot/dts/mediatek/mt8173-evb.dts
index 811cb76..d2cf4c5 100644
--- a/arch/arm64/boot/dts/mediatek/mt8173-evb.dts
+++ b/arch/arm64/boot/dts/mediatek/mt8173-evb.dts
@@ -191,6 +191,9 @@
 };
 
 &pwrap {
+	/* Only MT8173 E1 needs USB power domain */
+	power-domains = <&scpsys MT8173_POWER_DOMAIN_USB>;
+
 	pmic: mt6397 {
 		compatible = "mediatek,mt6397";
 		interrupt-parent = <&pio>;
-- 
1.9.1

